components/mediation-monitor/mediation-statistics/org.wso2.carbon.mediation.statistics/src/main/java/org/wso2/carbon/mediation/statistics/persistence/PersistingStatisticsObserver.java

```diff
@@ -102,16 +104,26 @@ public void notifyTraceLogs(MessageTraceLog[] logs) {
     }
 
     private void persistRecord(StatisticsRecord record) throws RegistryException {
-        String path = calculatePath(record);
-
-        Resource recordResource = registry.newResource();
-        recordResource.setProperty("totalCount", String.valueOf(record.getTotalCount()));
-        recordResource.setProperty("faultCount", String.valueOf(record.getFaultCount()));
-        recordResource.setProperty("maxTime", String.valueOf(record.getMaxTime()));
-        recordResource.setProperty("minTime", String.valueOf(record.getMinTime()));
-        recordResource.setProperty("avgTime", String.valueOf(record.getAvgTime()));
-        registry.put(path, recordResource);
-        recordResource.discard();
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext privilegedCarbonContext =
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext();
+
+            privilegedCarbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+            privilegedCarbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            String path = calculatePath(record);
+
+            Resource recordResource = registry.newResource();
+            recordResource.setProperty("totalCount", String.valueOf(record.getTotalCount()));

```

Comment:
Why do we have to explicitly run "String.valueOf" upon the attributes looked-up from an instance of "StatisticsRecord" class? Having each of the variables explicitly defined as Strings in the class definition would help you avoid similar instances.